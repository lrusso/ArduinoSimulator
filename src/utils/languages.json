{
  "en": {
    "EMPTY_SKETCH": "/*\n\nArduino Simulator 2 developed by LRusso.com\n\n*/\n\nvoid setup()\n\t{\n\tSerial.begin(9600);\n\t}\n\nvoid loop()\n\t{\n\t}",
    "FILENAME": "Untitled.ino",
    "FILENAME_EMPTY": "Untitled",
    "LOSECHANGES_TITLE": "MESSAGE",
    "LOSECHANGES_MESSAGE": "Lose any unsaved changes?",
    "LOSECHANGES_YES": "Yes",
    "LOSECHANGES_NO": "No",
    "DIGITAL_PINS": "DIGITAL PINS",
    "ANALOG_PINS": "ANALOG PINS",
    "SERIAL_MONITOR": "SERIAL MONITOR",
    "SEND": "SEND",
    "SEARCH_FOR": "Search for",
    "SEARCH_OF": " of ",
    "REPLACE_WITH": "Replace with",
    "REPLACE": "Replace",
    "ALL": "All"
  },
  "es": {
    "EMPTY_SKETCH": "/*\n\nArduino Simulator 2 desarrollado por LRusso.com\n\n*/\n\nvoid setup()\n\t{\n\tSerial.begin(9600);\n\t}\n\nvoid loop()\n\t{\n\t}",
    "FILENAME": "Sin nombre.ino",
    "FILENAME_EMPTY": "Sin nombre",
    "LOSECHANGES_TITLE": "MENSAJE",
    "LOSECHANGES_MESSAGE": "¿Perder cambios no guardados?",
    "LOSECHANGES_YES": "Sí",
    "LOSECHANGES_NO": "No",
    "DIGITAL_PINS": "PINES DIGITALES",
    "ANALOG_PINS": "PINES ANALÓGICOS",
    "SERIAL_MONITOR": "MONITOR SERIAL",
    "SEND": "ENVIAR",
    "SEARCH_FOR": "Buscar",
    "SEARCH_OF": " de ",
    "REPLACE_WITH": "Reemplazar con",
    "REPLACE": "Reemplazar",
    "ALL": "Todo"
  },
  "it": {
    "EMPTY_SKETCH": "/*\n\nArduino Simulator 2 sviluppato da LRusso.com\n\n*/\n\nvoid setup()\n\t{\n\tSerial.begin(9600);\n\t}\n\nvoid loop()\n\t{\n\t}",
    "FILENAME": "Senza titolo.ino",
    "FILENAME_EMPTY": "Senza titolo",
    "LOSECHANGES_TITLE": "MESSAGGIO",
    "LOSECHANGES_MESSAGE": "Perdere eventuali modifiche non salvate?",
    "LOSECHANGES_YES": "Sì",
    "LOSECHANGES_NO": "No",
    "DIGITAL_PINS": "PIN DIGITALI",
    "ANALOG_PINS": "PIN ANALOGICI",
    "SERIAL_MONITOR": "MONITOR SERIALE",
    "SEND": "SPEDIRE",
    "SEARCH_FOR": "Cercare",
    "SEARCH_OF": " di ",
    "REPLACE_WITH": "Sostituirlo con",
    "REPLACE": "Sostituirlo",
    "ALL": "Tutto"
  },
  "fr": {
    "EMPTY_SKETCH": "/*\n\nSimulateur Arduino 2 développé par LRusso.com\n\n*/\n\nvoid setup()\n\t{\n\tSerial.begin(9600);\n\t}\n\nvoid loop()\n\t{\n\t}",
    "FILENAME": "Sans titre.ino",
    "FILENAME_EMPTY": "Sans titre",
    "LOSECHANGES_TITLE": "MESSAGE",
    "LOSECHANGES_MESSAGE": "Perdre toutes les modifications non enregistrées?",
    "LOSECHANGES_YES": "Oui",
    "LOSECHANGES_NO": "Non",
    "DIGITAL_PINS": "BROCHES NUMÉRIQUES",
    "ANALOG_PINS": "BROCHES ANALOGIQUES",
    "SERIAL_MONITOR": "MONITEUR SÉRIE",
    "SEND": "ENVOYER",
    "SEARCH_FOR": "Rechercher",
    "SEARCH_OF": " sur ",
    "REPLACE_WITH": "Remplacer par",
    "REPLACE": "Remplacer",
    "ALL": "Toutes"
  },
  "pr": {
    "EMPTY_SKETCH": "/*\n\nArduino Simulator 2 desenvolvido por LRusso.com\n\n*/\n\nvoid setup()\n\t{\n\tSerial.begin(9600);\n\t}\n\nvoid loop()\n\t{\n\t}",
    "FILENAME": "Sem título.ino",
    "FILENAME_EMPTY": "Sem título",
    "LOSECHANGES_TITLE": "MENSAGEM",
    "LOSECHANGES_MESSAGE": "Perder alterações não salvas?",
    "LOSECHANGES_YES": "Sim",
    "LOSECHANGES_NO": "Não",
    "DIGITAL_PINS": "PINS DIGITAIS",
    "ANALOG_PINS": "PINS ANALÓGICOS",
    "SERIAL_MONITOR": "MONITOR SERIAL",
    "SEND": "MANDAR",
    "SEARCH_FOR": "Procurar",
    "SEARCH_OF": " de ",
    "REPLACE_WITH": "Substituir com",
    "REPLACE": "Substituir",
    "ALL": "Todos"
  }
}
