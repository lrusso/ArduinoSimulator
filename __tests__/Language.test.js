import React from "react"
import renderer from "react-test-renderer"
import languages from "../src/utils/languages.json"
import settings from "../src/utils/service.json"

describe("Language", () => {
  test("Checking the fallback language for missing strings", () => {
    let languageConflict = null

    Object.keys(languages).forEach((key) => {
      Object.keys(languages[key]).forEach((key2) => {
        if (languages[settings.LANGUAGE_FALLBACK][key2] === undefined) {
          languageConflict =
            settings.LANGUAGE_FALLBACK + " language is missing the string " + key2
        }
      })
    })

    expect(languageConflict).toBeNull()
  })

  test("Checking the other languages for missing strings", () => {
    let languageConflict = null

    Object.keys(languages[settings.LANGUAGE_FALLBACK]).forEach((keyOrig) => {
      Object.keys(languages).forEach((key) => {
        if (key === settings.LANGUAGE_FALLBACK) {
          return
        }
        Object.keys(languages[key]).forEach((key2) => {
          if (
            languages[settings.LANGUAGE_FALLBACK][key2] !== undefined &&
            languages[key][keyOrig] === undefined
          ) {
            languageConflict = key + " language is missing the string " + keyOrig
          }
        })
      })
    })

    expect(languageConflict).toBeNull()
  })

  test("Checking for empty strings", () => {
    let languageConflict = null

    Object.keys(languages).forEach((key) => {
      Object.keys(languages[key]).forEach((key2) => {
        if (languages[key][key2] === "") {
          languageConflict = key + " language - " + key2 + " value is empty"
        }
      })
    })

    expect(languageConflict).toBeNull()
  })

  test("Validating strings for possible changes", () => {
    let string_values = "\n"

    Object.keys(languages).forEach((key) => {
      Object.keys(languages[key]).forEach((key2) => {
        string_values = string_values + key + " - " + languages[key][key2] + "\n"
      })
    })

    const tree = renderer.create(<>{string_values}</>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
