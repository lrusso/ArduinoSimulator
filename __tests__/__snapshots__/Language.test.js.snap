// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Validating strings for possible changes 1`] = `
"
en - /*

Arduino Simulator 2 developed by LRusso.com

*/

void setup()
	{
	Serial.begin(9600);
	}

void loop()
	{
	}
en - Untitled.ino
en - Untitled
en - MESSAGE
en - Lose any unsaved changes?
en - Yes
en - No
en - DIGITAL PINS
en - ANALOG PINS
en - SERIAL MONITOR
en - SEND
en - Search for
en -  of 
en - Replace with
en - Replace
en - All
es - /*

Arduino Simulator 2 desarrollado por LRusso.com

*/

void setup()
	{
	Serial.begin(9600);
	}

void loop()
	{
	}
es - Sin nombre.ino
es - Sin nombre
es - MENSAJE
es - ¿Perder cambios no guardados?
es - Sí
es - No
es - PINES DIGITALES
es - PINES ANALÓGICOS
es - MONITOR SERIAL
es - ENVIAR
es - Buscar
es -  de 
es - Reemplazar con
es - Reemplazar
es - Todo
it - /*

Arduino Simulator 2 sviluppato da LRusso.com

*/

void setup()
	{
	Serial.begin(9600);
	}

void loop()
	{
	}
it - Senza titolo.ino
it - Senza titolo
it - MESSAGGIO
it - Perdere eventuali modifiche non salvate?
it - Sì
it - No
it - PIN DIGITALI
it - PIN ANALOGICI
it - MONITOR SERIALE
it - SPEDIRE
it - Cercare
it -  di 
it - Sostituirlo con
it - Sostituirlo
it - Tutto
fr - /*

Simulateur Arduino 2 développé par LRusso.com

*/

void setup()
	{
	Serial.begin(9600);
	}

void loop()
	{
	}
fr - Sans titre.ino
fr - Sans titre
fr - MESSAGE
fr - Perdre toutes les modifications non enregistrées?
fr - Oui
fr - Non
fr - BROCHES NUMÉRIQUES
fr - BROCHES ANALOGIQUES
fr - MONITEUR SÉRIE
fr - ENVOYER
fr - Rechercher
fr -  sur 
fr - Remplacer par
fr - Remplacer
fr - Toutes
pr - /*

Arduino Simulator 2 desenvolvido por LRusso.com

*/

void setup()
	{
	Serial.begin(9600);
	}

void loop()
	{
	}
pr - Sem título.ino
pr - Sem título
pr - MENSAGEM
pr - Perder alterações não salvas?
pr - Sim
pr - Não
pr - PINS DIGITAIS
pr - PINS ANALÓGICOS
pr - MONITOR SERIAL
pr - MANDAR
pr - Procurar
pr -  de 
pr - Substituir com
pr - Substituir
pr - Todos
"
`;
